ui/pages/reservation-unit/[id].tsx

```diff
@@ -402,6 +413,13 @@ const ReservationUnit = ({
     [reservationUnit.canApplyFreeOfCharge]
   );
 
+  useEffect(() => {
+    setFocusDate(new Date());
+    setInitialReservation(null);
+    setIsDialogOpen(false);
+    setPendingReservation(null);
+  }, [dynamicRoute, setPendingReservation, setReservation]);
```

Comment:
Why is this needed? is there a storage that persists through routes? e.g. context
Wouldn't the right solution to be to clear the context on these conditions? or at least wrap it into a named hook so it's clear why and what it does.

Reply:
Turned out that nextjs persists dynamic page state across route changes. What actually helped was the `key` prop to the component/page and the other cleanup was unnecessary.